components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/SQLQueries.java

```diff
@@ -197,6 +197,10 @@
             "IDN_OAUTH2_AUTHORIZATION_CODE WHERE CONSUMER_KEY_ID IN (SELECT ID FROM IDN_OAUTH_CONSUMER_APPS WHERE " +
             "CONSUMER_KEY = ?) AND STATE = ?";
 
+    public static final String UPDATE_AUTHORIZATION_CODE_STATE_FOR_CONSUMER_KEY = "UPDATE "
+            + "IDN_OAUTH2_AUTHORIZATION_CODE SET STATE=? WHERE CONSUMER_KEY_ID IN (SELECT ID FROM "
```

Comment:
parameter substitution happens when constructing the query. So, the comparatively the gain is low. Further more, this method is useful later when we want to bulk update the AUTH codes.